contracts/tokenbridge/libraries/gateway/GatewayRouter.sol

```diff
@@ -144,4 +157,59 @@ abstract contract GatewayRouter is TokenGateway, IGatewayRouter {
         }
         return TokenGateway(gateway).calculateL2TokenAddress(l1ERC20);
     }
+
+    /**
+     * @notice Bridge ERC20 token using the registered or otherwise default gateway with call to permit. 
+                Compatible with older gateways without OutboundTransferCustomRefund
+     * @param _token L1 address of ERC20
+     * @param _to Account to be credited with the tokens in the L2 (can be the user's L2 account or a contract), not subject to L2 aliasing
+                  This account, or its L2 alias if it have code in L1, will also be able to cancel the retryable ticket and receive callvalue refund
+     * @param _amount Token Amount
+     * @param _maxGas Max gas deducted from user's L2 balance to cover L2 execution
+     * @param _gasPriceBid Gas price for L2 execution
+     * @param _data encoded data from router and user
+     * @param _permitData signature and deadline params of permit
+    */
+    function outboundTransferWithPermit(
+        address _token,
+        address _to,
+        uint256 _amount,
+        uint256 _maxGas,
+        uint256 _gasPriceBid,
+        bytes calldata _data,
+        PermitData calldata _permitData
+    ) public payable virtual returns (bytes memory) {
+        callPermit(_token, _amount, _permitData);
+        return outboundTransfer(_token, _to, _amount, _maxGas, _gasPriceBid, _data);
+    }
+
+    function callPermit(
+        address _token,
+        uint256 _amount,
+        PermitData calldata _permitData
+    ) internal virtual {
+        address gateway = getGateway(_token);
+        if (_permitData.isStandardImpl) {
```

Comment:
I think we could implement a separate entrypoint function instead of `bool isStandardImpl`
something like `outboundTransferWithEip2612Permit` and `outboundTransferWithDaiPermit`
would save us a bit of gas